src/sage/algebras/quatalg/quaternion_algebra.py

```diff
@@ -403,50 +403,61 @@ def is_commutative(self) -> bool:
-        quaternion algebra is isomorphic to the 2x2 matrix algebra.
+        Check whether this quaternion algebra is a division algebra,
+        i.e. whether every nonzero element in it is invertible.
 
         EXAMPLES::
 
             sage: QuaternionAlgebra(QQ,-5,-2).is_division_algebra()
             True
             sage: QuaternionAlgebra(1).is_division_algebra()
             False
-            sage: QuaternionAlgebra(2,9).is_division_algebra()
-            False
+
+            sage: K = QuadraticField(3)
+            sage: L = QuadraticField(-15)
+            sage: QuaternionAlgebra(K, -1, -1).is_division_algebra()
+            True
+            sage: QuaternionAlgebra(L, -1, -1).is_division_algebra()
+            True
+
             sage: QuaternionAlgebra(RR(2.),1).is_division_algebra()
             Traceback (most recent call last):
             ...
-            NotImplementedError: base field must be rational numbers
+            NotImplementedError: base field must be rational numbers or a number field
         """
-        if not is_RationalField(self.base_ring()):
-            raise NotImplementedError("base field must be rational numbers")
-        return self.discriminant() != 1
+        try:
+            return self.ramified_places(inf=True) != ([], [])
+        except ValueError:
+            raise NotImplementedError("base field must be rational numbers or a number field")
 
     def is_matrix_ring(self) -> bool:
         """
-        Return ``True`` if the quaternion algebra is isomorphic to the 2x2
-        matrix ring, and ``False`` if ``self`` is a division algebra (i.e.
-        every nonzero element in ``self`` is invertible).
+        Check whether this quaternion algebra is isomorphic to the
+        2x2 matrix ring over the base field.
 
         EXAMPLES::
 
             sage: QuaternionAlgebra(QQ,-5,-2).is_matrix_ring()
             False
-            sage: QuaternionAlgebra(1).is_matrix_ring()
-            True
             sage: QuaternionAlgebra(2,9).is_matrix_ring()
             True
+
+            sage: K = QuadraticField(3)
+            sage: L = QuadraticField(-15)
+            sage: QuaternionAlgebra(K, -1, -1).is_matrix_ring()
+            False
+            sage: QuaternionAlgebra(L, -1, -1).is_matrix_ring()
+            False
+
             sage: QuaternionAlgebra(RR(2.),1).is_matrix_ring()
             Traceback (most recent call last):
             ...
-            NotImplementedError: base field must be rational numbers
-
+            NotImplementedError: base field must be rational numbers or a number field
         """
-        if not is_RationalField(self.base_ring()):
-            raise NotImplementedError("base field must be rational numbers")
-        return self.discriminant() == 1
+        try:
+            return self.ramified_places(inf=True) == ([], [])
+        except ValueError:
+            raise NotImplementedError("base field must be rational numbers or a number field")
```

Comment:
Correct me if I'm wrong, but isn't this just `not self.is_division_algebra()`?

Reply:
Yes, it is - I implemented it independently because that's how it's also done in the code right now. Do you think I should just replace one by a call to the other?

Reply:
Yeah avoid code duplication

Reply:
There is a small caveat here, namely that these two methods are probably only opposite to each other as long as the base ring is a field - but since they were never implemented for any non-field, I'll change it as you suggested.

Reply:
If so, don't replace the entire 4 lines with `not ...`, but only the `try`-ed case (and maybe add a comment noting this ^). This way reduces the chance the other person makes a mistake, I guess

Reply:
In `.is_division_algebra()` I catch a `ValueError` and raise an `NotImplementedError`; but if I would try to catch a `NotImplementedError` in `.is_matrix_ring()`, this might cause errors if the `.is_division_algebra()`-method gets extended first to a case where it is not the opposite of `.is_matrix_ring()` anymore; so maybe it makes to leave it as is for now, since it would only replace one line anyway?

Reply:
Yeah whatever smzall issues